tests/conftest.py

```diff
@@ -18,9 +18,9 @@
 async def worker(arq_redis):
     worker_: Worker = None
 
-    def create(functions=[], burst=True, poll_delay=0, **kwargs):
+    def create(functions=[], burst=True, poll_delay=0, queue_read_limit=40, **kwargs):
         nonlocal worker_
-        worker_ = Worker(functions=functions, redis_pool=arq_redis, burst=burst, poll_delay=poll_delay, **kwargs)
+        worker_ = Worker(functions=functions, redis_pool=arq_redis, burst=burst, poll_delay=poll_delay, queue_read_limit=queue_read_limit, **kwargs)
```

Comment:
The only thing causing tests to fail on CI is this line which is too long, please run `make format`.

Reply:
@samuelcolvin Thanks, done. On my system two tests, `test_no_jobs` and `test_health_check_direct` always fail. I don't know the reason, but this is the error in both cases:

```
self = <asyncio.unix_events._UnixDefaultEventLoopPolicy object at 0x7fc091335e10>

    def get_event_loop(self):
        """Get the event loop.
    
        This may be None or an instance of EventLoop.
        """
        if (self._local._loop is None and
                not self._local._set_called and
                isinstance(threading.current_thread(), threading._MainThread)):
            self.set_event_loop(self.new_event_loop())
    
        if self._local._loop is None:
            raise RuntimeError('There is no current event loop in thread %r.'
>                              % threading.current_thread().name)
E           RuntimeError: There is no current event loop in thread 'MainThread'.

/usr/lib64/python3.7/asyncio/events.py:644: RuntimeError
```

Reply:
humm, not sure, seems to work fine for me:

```
(env) arq 0  11ms ➤  pytest -k test_health_check_direct
Test session starts (platform: linux, Python 3.7.4, pytest 4.4.1, pytest-sugar 0.9.2)
rootdir: /home/samuel/code/arq, inifile: setup.cfg, testpaths: tests
plugins: mock-1.10.4, toolbox-0.4, cov-2.6.1, timeout-1.3.3, sugar-0.9.2, aiohttp-0.3.0
timeout: 5.0s
timeout method: signal
timeout func_only: False
collecting ... 
 tests/test_worker.py ✓                                          100% ██████████

Results (0.06s):
       1 passed
      97 deselected
(env) arq 0 419ms ➤  
```

Do you have the same plugins and versions?

Reply:
@samuelcolvin I have no idea why it does not work. I have the exact same version of everything, even Python. It should be tracked in a new issue, because I get the same on `master`.

```
 pytest -k test_health_check_direct
Test session starts (platform: linux, Python 3.7.4, pytest 4.4.1, pytest-sugar 0.9.2)
rootdir: /home/miki/exp/arq, inifile: setup.cfg, testpaths: tests
plugins: mock-1.10.4, sugar-0.9.2, timeout-1.3.3, cov-2.6.1, aiohttp-0.3.0, toolbox-0.4
timeout: 5.0s
timeout method: signal
timeout func_only: False
collecting ... 

―――――――――――――――――――――――――――――――――――― test_health_check_direct ――――――――――――――――――――――――――――――――――――

loop = <_UnixSelectorEventLoop running=False closed=False debug=False>

    def test_health_check_direct(loop):
        class Settings:
            pass
    
>       assert check_health(Settings) == 1

tests/test_worker.py:41: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
arq/worker.py:610: in check_health
    loop = asyncio.get_event_loop()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <asyncio.unix_events._UnixDefaultEventLoopPolicy object at 0x7f080bc70210>

    def get_event_loop(self):
        """Get the event loop.
    
        This may be None or an instance of EventLoop.
        """
        if (self._local._loop is None and
                not self._local._set_called and
                isinstance(threading.current_thread(), threading._MainThread)):
            self.set_event_loop(self.new_event_loop())
    
        if self._local._loop is None:
            raise RuntimeError('There is no current event loop in thread %r.'
>                              % threading.current_thread().name)
E           RuntimeError: There is no current event loop in thread 'MainThread'.

/usr/lib64/python3.7/asyncio/events.py:644: RuntimeError

 tests/test_worker.py ⨯                                                            100% ██████████

Results (0.31s):
       1 failed
         - tests/test_worker.py:37 test_health_check_direct
      97 deselected
```

Reply:
new issue please.

This won't be related to pytest-asyncio, since we use pytest-aiohttp for async tests.

I also doubt it's related to the aiohttp issue since that's quite old, the event loop logic changed in `3.5.3` and most libraries have gradually changed how they work with `loop` since they've ceased to support `<=3.5.2`.

Reply:
I have opened issue #144. Note that in the pytest-asyncio issue pytest-aiohttp is mentioned.